[CI] Zen1IT.testFreshestMasterElectedAfterFullClusterRestart failure on 7.x

CI link: https://scans.gradle.com/s/labarsogt5yjy/console-log?task=:server:integTest

Reproduce line, does not reproduce locally:
```
./gradlew :server:integTest --tests "org.elasticsearch.cluster.coordination.Zen1IT.testFreshestMasterElectedAfterFullClusterRestart" \
  -Dtests.seed=762050F5C7A4616A \
  -Dtests.security.manager=true \
  -Dtests.locale=sr-ME \
  -Dtests.timezone=CST \
  -Dcompiler.java=12 \
  -Druntime.java=8
```

Stacktrace:
```
2> јул 16, 2019 5:39:45 AM com.carrotsearch.randomizedtesting.RandomizedRunner$QueueUncaughtExceptionsHandler uncaughtException
--
2> WARNING: Uncaught exception in thread: Thread[elasticsearch[node_t2][generic][T#2],5,TGRP-Zen1IT]
2> java.lang.AssertionError: Interruption via Thread#interrupt() is unsupported. Use CancellableThreads#cancel() instead
2> 	at __randomizedtesting.SeedInfo.seed([762050F5C7A4616A]:0)
2> 	at org.elasticsearch.common.util.CancellableThreads.executeIO(CancellableThreads.java:111)
2> 	at org.elasticsearch.common.util.CancellableThreads.execute(CancellableThreads.java:89)
2> 	at org.elasticsearch.discovery.SeedHostsResolver.resolveHostsLists(SeedHostsResolver.java:148)
2> 	at org.elasticsearch.discovery.zen.UnicastZenPing.lambda$createHostsResolver$0(UnicastZenPing.java:152)
2> 	at org.elasticsearch.discovery.FileBasedSeedHostsProvider.getSeedAddresses(FileBasedSeedHostsProvider.java:78)
2> 	at org.elasticsearch.discovery.DiscoveryModule.lambda$new$4(DiscoveryModule.java:142)
2> 	at org.elasticsearch.discovery.zen.UnicastZenPing.ping(UnicastZenPing.java:198)
2> 	at org.elasticsearch.discovery.zen.UnicastZenPing.ping(UnicastZenPing.java:187)
2> 	at org.elasticsearch.discovery.zen.ZenDiscovery.pingAndWait(ZenDiscovery.java:962)
2> 	at org.elasticsearch.discovery.zen.ZenDiscovery.findMaster(ZenDiscovery.java:806)
2> 	at org.elasticsearch.discovery.zen.ZenDiscovery.innerJoinCluster(ZenDiscovery.java:442)
2> 	at org.elasticsearch.discovery.zen.ZenDiscovery.access$1800(ZenDiscovery.java:94)
2> 	at org.elasticsearch.discovery.zen.ZenDiscovery$JoinThreadControl$1.run(ZenDiscovery.java:1173)
2> 	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:703)
2> 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
2> 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
2> 	at java.lang.Thread.run(Thread.java:748)
```

Not muting since it appears sporadically.

Full logs can be found [here](https://github.com/elastic/elasticsearch/files/3396998/console.log.zip)

--- a/server/src/main/java/org/elasticsearch/discovery/zen/UnicastZenPing.java
+++ b/server/src/main/java/org/elasticsearch/discovery/zen/UnicastZenPing.java
@@ -101,8 +101,6 @@ public class UnicastZenPing implements ZenPing {
 
     private final AtomicInteger pingingRoundIdGenerator = new AtomicInteger();
 
-    private final CancellableThreads cancellableThreads = new CancellableThreads();
-
     private final Map<Integer, PingingRound> activePingingRounds = newConcurrentMap();
 
     // a list of temporal responses a node will return for a request (holds responses from other nodes)
@@ -149,13 +147,21 @@ public class UnicastZenPing implements ZenPing {
     }
 
     private SeedHostsProvider.HostsResolver createHostsResolver() {
-        return hosts -> SeedHostsResolver.resolveHostsLists(cancellableThreads, unicastZenPingExecutorService, logger, hosts,
-            transportService, resolveTimeout);
+        return hosts -> SeedHostsResolver.resolveHostsLists(
+            new CancellableThreads() {
+                public void execute(Interruptible interruptible) {
+                    try {
+                        interruptible.run();
+                    } catch (InterruptedException e) {
+                        throw new CancellableThreads.ExecutionCancelledException("interrupted by " + e);
+                    }
+                }
+            },
+            unicastZenPingExecutorService, logger, hosts, transportService, resolveTimeout);
     }
 
     @Override
     public void close() {
-        cancellableThreads.cancel("stopping UnicastZenPing");
         ThreadPool.terminate(unicastZenPingExecutorService, 10, TimeUnit.SECONDS);
         Releasables.close(activePingingRounds.values());
         closed = true;

